cpp/open3d/core/kernel/ReductionCPU.cpp

```diff
@@ -190,14 +191,25 @@ class CPUArgReductionEngine {
         // elements. We need to keep track of the indices within each
         // sub-iteration.
         int64_t num_output_elements = indexer_.NumOutputElements();
+        if (num_output_elements <= 1) {
+            LaunchArgReductionKernelTwoPass(indexer_, reduce_func, identity);
+        } else {
+            LaunchArgReductionParallelDim(indexer_, reduce_func, identity);
+        }
+    }
 
+    template <typename scalar_t, typename func_t>
+    static void LaunchArgReductionParallelDim(const Indexer& indexer,
+                                              func_t reduce_func,
+                                              scalar_t identity) {
+        int64_t num_output_elements = indexer.NumOutputElements();
 #pragma omp parallel for schedule(static) \
         num_threads(utility::EstimateMaxThreads())
         for (int64_t output_idx = 0; output_idx < num_output_elements;
```

Comment:
Since usually `num_output_elements << num_threads`, we should collapse the two for loops to improve thread utilization with`#pragma omp for collapse(2)` I believe Visual Studio now supports this (OpenMP v3) but good to check Windows CI.

Reply:
I just want to remind about #6626 as this will add new OpenMP code

Reply:
When simply launching `LaunchArgReductionKernelTwoPass` inside the outer for loop, it improves the performance when reducing over the large_dim, but worsens the performance of reducing over the small dim. Therefore, this would require a more sophisticated way to determine how many threads we should launch in the inner loop.

The image shows the time, so `+4.5` means this approach would be 4.5 times slower in that benchmark, 
`-0.78` means roughly 4.5 times faster 
![image](https://github.com/user-attachments/assets/14085955-8705-4e64-b9f0-ccc9af330dfb)

Reply:
Thanks @manuelvogel12 for the detailed benchmarking. This looks good to me.

To address @benjaminum's point, we will be switching from OpenMP to TBB soon (PR #6626). This helps quite a bit with lack of good support for OpenMP on Windows and Apple, simplifies the code and also comes with the benefit of composable parallelism. We also expect (some) speedup from TBB's work stealing scheduler model.

I think we can merge this PR due to its speed boost for now and then switch this code to TBB as well along with the rest of the code. @manuelvogel12 I would encourage you to check out PR #6626 and help us test that.